### PR TITLE
mypy.ini: Globally enable `disallow_untyped_defs = True`

### PR DESCRIPTION
**Pull request by sebastian-philipp**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/ceph_module.pyi [hunk 1]
```diff
@@ -67,7 +67,7 @@ class BaseMgrModule(object):
     def _ceph_lookup_release_name(self, release: int) -> str: ...
     def _ceph_cluster_log(self, channel: str, priority: int, message: str) -> None: ...
     def _ceph_get_context(self) -> object: ...
-    def _ceph_get(self, data_name: str) -> Dict[str, Any]: ...
+    def _ceph_get(self, data_name: str) -> Any: ...
```
Comment (tchaikov) [inline src/pybind/mgr/ceph_module.pyi:70]: why? i think it is correct.

**Comment by tchaikov** [comment]
```
mypy run-test: commands[0] | mypy --config-file ../mypy.ini cephadm
cephadm: note: In member "__init__" of class "BaseConfig":
cephadm:107: error: Function is missing a return type annotation
cephadm:107: note: Use "-> None" if function does not return a value
cephadm: note: In member "set_from_args" of class "BaseConfig":
cephadm:126: error: Function is missing a return type annotation
cephadm: note: In member "__init__" of class "CephadmContext":
cephadm:135: error: Function is missing a return type annotation
```

we are not quite close =(

File: src/pybind/mgr/ceph_module.pyi (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/ceph_module.pyi:70]: `pg_ready` dumps a bool:

https://github.com/ceph/ceph/blob/5c9a3962b76374e2d94a7ef350a17dee8598b290/src/mgr/ActivePyModules.cc#L380-L383

**Comment by sebastian-philipp** [comment]
> ```
> mypy run-test: commands[0] | mypy --config-file ../mypy.ini cephadm
> cephadm: note: In member "__init__" of class "BaseConfig":
> cephadm:107: error: Function is missing a return type annotation
> cephadm:107: note: Use "-> None" if function does not return a value
> cephadm: note: In member "set_from_args" of class "BaseConfig":
> cephadm:126: error: Function is missing a return type annotation
> cephadm: note: In member "__init__" of class "CephadmContext":
> cephadm:135: error: Function is missing a return type annotation
> ```
> 
> we are not quite close =(

done!

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
adding needs-qa, just in case.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-08-04_05:19:17-rados-wip-kefu-testing-2021-08-04-1022-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-08-04_10:33:19-rados-wip-kefu-testing-2021-08-04-1022-distro-basic-smithi/

**Pull request merged**